### PR TITLE
NumberBox: The caret jumps to the beginning of the input after the minus sign is typed if the format is '#' (T1093688)

### PR DESCRIPTION
**Pull request by EugeniyKiyashko**

### CODE REVIEW

File: js/localization/number.js [hunk 1]
```diff
@@ -257,6 +257,10 @@ const numberLocalization = dependencyInjector({
             return 1;
         }
 
+        if(text === '-') {
```
Comment (Zedwag) [inline js/localization/number.js]: can we fix this in numberBox files? It doesn't seem good to change core squad files t fix such a little bug

File: testing/tests/DevExpress.ui.widgets.editors/numberBoxParts/mask.tests.js [hunk 2]
```diff
@@ -2093,15 +2093,19 @@ QUnit.module('stubs', moduleConfig, function() {
...
+    ].forEach(({ format, expectedText, typedText, expectedValue }) => {
         QUnit.test(`widget should correctly apply format="${format}", value="${expectedValue}"`, function(assert) {
             this.instance.option({
                 value: null,
                 format
             });
 
             this.keyboard
-                .type(expectedValue.toString())
+                .type(typedText || expectedValue.toString())
```
Comment (ksercs) [inline testing/tests/DevExpress.ui.widgets.editors/numberBoxParts/mask.tests.js:2119]:
```suggestion
                .type(typedText ?? expectedValue.toString())
```

File: js/localization/number.js (hunk 1 above)
Comment (ksercs) [inline js/localization/number.js]: Hm, it works only for simple formats, it will not work for `#;-#` for example, or `0;<0`. Moreover it's not the best practice to add such corner case. Maybe it's better to update etalon calculation? Seems like we can just modify 247 row this way:
`negativeEtalon = negativeEtalon.replaceAll('1', '.*');`

File: testing/tests/DevExpress.ui.widgets.editors/numberBoxParts/mask.tests.js (hunk 2 above)
Comment (EugeniyKiyashko) [reply testing/tests/DevExpress.ui.widgets.editors/numberBoxParts/mask.tests.js:2119]: ?? doesn't supported in our test environment

File: js/localization/number.js (hunk 1 above)
Comment (EugeniyKiyashko) [reply js/localization/number.js]: https://github.com/DevExpress/DevExtreme/pull/21806/commits/5501331453ded0b91d021e49fa43b142b22afce6

File: testing/tests/DevExpress.ui.widgets.editors/numberBoxParts/mask.tests.js [hunk 3]
```diff
@@ -2093,15 +2093,30 @@ QUnit.module('stubs', moduleConfig, function() {
...
+        { format: '\'--\'#', expectedText: '---1', typedText: '-1', expectedValue: -1 },
+        { format: '#;-#', expectedText: '-1', typedText: '1-', expectedValue: -1 },
+        { format: '#;-#', expectedText: '1', typedText: '1--', expectedValue: 1 },
+        { format: '0;<0', expectedText: '<1', typedText: '-1', expectedValue: -1 },
+        { format: '0;<0', expectedText: '<1', typedText: '1-', expectedValue: -1 },
+        { format: '0;<0', expectedText: '1', typedText: '1--', expectedValue: 1 },
+        { format: '\'plus\' 0;\'minus\' 0', expectedText: 'minus 1', typedText: '1-', expectedValue: -1 },
+        { format: '\'plus\' 0;\'minus\' 0', expectedText: 'plus 1', typedText: '1--', expectedValue: 1 },
+    ].forEach(({ format, expectedText, typedText, expectedValue }) => {
+        QUnit.test(`widget should correctly apply format="${format}", value="${typedText || expectedValue}"`, function(assert) {
```
Comment (ksercs) [inline testing/tests/DevExpress.ui.widgets.editors/numberBoxParts/mask.tests.js:2112]: nice work!

**Review by ksercs — APPROVED** [review]

**Review by Zedwag — APPROVED** [review]

File: js/localization/number.js (hunk 1 above)
Comment (EugeniyKiyashko) [reply js/localization/number.js]: Before fix the single char '-' doesn't determined as possible negative value, but before regression it was.

File: testing/tests/DevExpress.ui.widgets.editors/numberBoxParts/mask.tests.js (hunk 3 above)
Comment (EugeniyKiyashko) [reply testing/tests/DevExpress.ui.widgets.editors/numberBoxParts/mask.tests.js:2112]: you are welcome!
